src/main/java/space/obminyashka/items_exchange/controller/AuthController.java

```diff
@@ -114,6 +125,8 @@ public ResponseEntity<String> registerUser(@RequestBody @Valid UserRegistrationD
 
         if (userService.registerNewUser(userRegistrationDto)) {
             log.info("User with email: {} successfully registered", escapeHtml(userRegistrationDto.getEmail()));
+            token.setUser(userService.findByUsernameOrEmail(userRegistrationDto.getEmail()).get());
```

Comment:
That is excess call to the DB. Would you mind using existing logic or update it for your needs instead of making an additional call?